Add PIM PAG API documentation in graph beta (#19585)
microsoftgraph/microsoft-graph-docs: api-reference/beta/api/privilegedaccessgroupassignmentschedule-filterbycurrentuser.md

---
title: "privilegedAccessGroupAssignmentSchedule: filterByCurrentUser"

> description: "Retrieve the schedules of privileged access to a group relationship assignment for which the signed-in user is the principal."

Reviewer: I've only tweaked the suggestion in the "request" filterByCurrentUser doc.
Suggestion to avoid "signed-in" because there's no sign in in app-scenarios.
```suggestion
description: "In PIM for groups, retrieve the schedules for access assignments for the calling principal to groups that are governed by PIM."
```

author: "mandardalvi"
ms.localizationpriority: medium
ms.prod: "governance"
doc_type: apiPageType
---

# privilegedAccessGroupAssignmentSchedule: filterByCurrentUser
Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> Retrieve the schedules of privileged access to a group relationship assignment for which the signed-in user is the principal.

Reviewer: Match with Line 3

## Permissions
One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

|Permission type|Permissions (from least to most privileged)|
|:---|:---|

> |Delegated (work or school account)|PrivilegedAccess.ReadWrite.AzureADGroup, PrivilegedAccess.Read.AzureADGroup|

Reviewer: Proper order

|Delegated (personal Microsoft account)|Not supported.|
|Application|PrivilegedAccess.ReadWrite.AzureADGroup, PrivilegedAccess.Read.AzureADGroup|

> 

Reviewer: Add RBAC

## HTTP request

<!-- {
  "blockType": "ignored"
}
-->
``` http

> GET /identityGovernance/privilegedAccess/group/assignmentSchedules/filterByCurrentUser

Reviewer: Same as the request filterByCurrentUser API - I believe the function parameters are mandatory. Therefore, update this URL to add the function parameters.

```

## Function parameters
In the request URL, provide the following query parameters with values.
The following table shows the parameters that can be used with this function.

|Parameter|Type|Description|
|:---|:---|:---|
|on|assignmentScheduleFilterByCurrentUserOptions|Filter used to query assignmentSchedules. The possible values are `principal`, `unknownFutureValue`. Required.|

## Request headers
|Name|Description|
|:---|:---|
|Authorization|Bearer {token}. Required.|

## Request body
Do not supply a request body for this method.

## Response

If successful, this function returns a `200 OK` response code and a [privilegedAccessGroupAssignmentSchedule](../resources/privilegedaccessgroupassignmentschedule.md) collection in the response body.

## Examples

### Request
The following is an example of a request.
<!-- {
  "blockType": "request",
  "name": "privilegedaccessgroupassignmentschedulethis.filterbycurrentuser"
}
-->
``` http

> GET https://graph.microsoft.com/beta/identityGovernance/privilegedAccess/group/assignmentSchedules/filterByCurrentUser(on='parameterValue')

Reviewer: Replace "parameterValue" with an actual value - In this cause only `principal` is supported.

```

### Response
The following is an example of the response.
>**Note:** The response object shown here might be shortened for readability.
<!-- {
  "blockType": "response",
  "truncated": true,
  "@odata.type": "Collection(microsoft.graph.privilegedAccessGroupAssignmentSchedule)"
}
-->
``` http
HTTP/1.1 200 OK
Content-Type: application/json

> {

Reviewer: Replace with payload that matches the updated request URL above
